### PR TITLE
Add an umbrella import to the `_TestingInternals` module map file.

### PR DESCRIPTION
**Pull request by grynspan**
This PR removes the individual header declarations from the `_TestingInternals` module map file and replaces them with a single umbrella import. This fixes an existing workflow used by some of our teammates.

~~This PR moves the module map file to align with an existing workflow used by some of our teammates.~~

### Checklist:

- [x] Code and documentation should follow the style of the [Style Guide](https://github.com/apple/swift-testing/blob/main/Documentation/StyleGuide.md).
- [x] If public symbols are renamed or modified, DocC references should be updated.

### CODE REVIEW

**Comment by grynspan** [comment]
@swift-ci please test

File: Sources/_TestingInternals/module.modulemap
Comment (stmontgomery) [inline Sources/_TestingInternals/module.modulemap:1]: If you move this I imagine the CMake build may need to be adjusted

Comment (grynspan) [reply Sources/_TestingInternals/module.modulemap:1]: I asked Jeff/Saleem to verify I didn't break their changes.

**Comment by grynspan** [comment]
> Are you sure that, when a module map is placed in this location, SwiftPM actually honors it and does not revert back to its default behavior of synthesizing a module map representing all the headers in `include`?

It doesn't actually matter. SwiftPM is fine to synthesize a module here. The module map is only used by CMake and the Xcode project generated by it.

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
Windows failure is a known compiler issue.

**Comment by jeffdav** [comment]
I'm on pto but will verify tomorrow!

**Comment by compnerd** [comment]
Tested 004994fef30548d6370dd141dfbb5154cb5bc59c with the latest build from https://github.com/thebrowsercompany/swift-build as the current swift.org toolchains are too old to support the package at this point (due to the use of the stdlib macros).

**Comment by grynspan** [comment]
Curious. I wonder what's confusing CMake there given that there aren't actually any source changes involved.

**Comment by compnerd** [comment]
It's unclear if this change triggered this, just something that I noticed :)

**Pull request merged**